src/pypeh/core/interfaces/outbound/dataops.py

```diff
@@ -130,6 +132,65 @@ def validate(
         return ret
 
 
+class DataEnrichmentInterface(OutDataOpsInterface, Generic[T_DataType]):
+    @abstractmethod
+    def _enrich_data(self, data: dict[str, Sequence] | T_DataType, config: dict) -> T_DataType:
+        raise NotImplementedError
+
+    @classmethod
+    def get_default_adapter_class(cls):
+        try:
+            adapter_module = importlib.import_module("pypeh.adapters.outbound.enrichment.dataops")
+            adapter_class = getattr(adapter_module, "DataFrameAdapter")
+        except Exception as e:
+            logger.error("Exception encountered while attempting to import enrichment DataFrameAdapter")
+            raise e
+        return adapter_class
+
+    @staticmethod
+    def _transform_source_path(observation_id: str, source_path: str) -> str:
+        if "\\" in source_path:
+            return source_path
+        else:
+            return f"{observation_id}\\{source_path}"
+
+    @staticmethod
+    def _get_parents_nodes(calculation_designs: list[CalculationDesign | None]) -> list[str]:
```

Comment:
just being picky but this does not create parent nodes, it extracts `CalculationKeywordArgument`s from a `CalculationDesign`